### PR TITLE
Page carousel component

### PR DESCRIPTION
**Pull request by vidyanambiar**
Adds a `PageCarousel` to the shared UI framework. The component displays cards within it and switches smoothly between pages of cards. The carousel is responsive and displays a certain number of cards within it based on the screen size.

https://github.com/ansible/ansible-ui/assets/43621546/a4635326-5d39-4384-8bfd-e050df4becfb

### CODE REVIEW

**Review by jamestalton — APPROVED** [review]
lgtm

File: framework/PageCarousel/PageCarouselNav.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { Button, Bullseye, ButtonVariant } from '@patternfly/react-core';
+import { AngleLeftIcon, AngleRightIcon } from '@patternfly/react-icons';
+import { CSSProperties, useEffect } from 'react';
+import { useMemo } from 'react';
+
+const pageDotStyle: CSSProperties = {
```
Comment (keithjgrant) [inline framework/PageCarousel/PageCarouselNav.tsx:6]: This could be converted to styled components to match our pattern elsewhere

**Review by keithjgrant — APPROVED** [review]

File: framework/PageCarousel/PageCarouselNav.tsx (hunk 1 above)
Comment (vidyanambiar) [reply framework/PageCarousel/PageCarouselNav.tsx:6]: @keithjgrant I had tried this but couldn't get the dots to render. And the custom styles are not getting applied to the PF Button. The other places that I used it I was not modifying a PF component. Could you look this code over please?

```
const PageNavDotButton = styled(Button)`
  height: 10;
  width: 10;
  border-radius: 50%;
  padding: 0;
  margin-left: 10;
  margin-right: 10;
  ${(props: { isCurrentPage: boolean }) =>
    props.isCurrentPage
      ? css`
          background-color: var(--pf-global--primary-color--100);
        `
      : css`
          background-color: var(--pf-global--disabled-color--200);
        `}
`;

...

  const pageDots = useMemo(() => {
    const dots = [];
    for (let i = 0; i < totalPages; i++) {
      dots.push(
        <PageNavDotButton
          key={i}
          variant={ButtonVariant.plain}
          onClick={() => {
            setPage(i);
          }}
          isCurrentPage={i === currentPage}
        ></PageNavDotButton>
      );
    }
    return dots;
  }, [currentPage, setPage, totalPages]);

```

image

Comment (vidyanambiar) [reply framework/PageCarousel/PageCarouselNav.tsx]: @keithjgrant Thanks for helping me resolve this! I've updated the PR

**Pull request merged**